usr/src/uts/intel/io/vmxnet3s/vmxnet3_tx.c

```diff
@@ -155,30 +126,32 @@ vmxnet3_tx_prepare_offload(vmxnet3_softc_t *dp, vmxnet3_offload_t *ol,
 
 			ol->om = VMXNET3_OM_TSO;
 			ol->hlen = ethLen + ipLen + tcpLen;
-			ol->msscof = DB_LSOMSS(mp);
+			ol->msscof = mss;
 
 			if (mblk != mp) {
 				ret = ol->hlen;
 			}
+		} else if (flags & HCK_PARTIALCKSUM) {
```

Comment:
I'm assuming that vmxnet3 LSO also does the full hardware checksum, and that this is why VMware made this change. Given that we have no virtual hardware spec for vmxnet3, I can't be 100% sure.